packages/sanity/src/structure/panes/document/useDocumentPaneInspector.ts

```diff
@@ -0,0 +1,192 @@
+import {omit} from 'lodash'
+import {useCallback, useEffect, useMemo, useRef, useState} from 'react'
+import {type DocumentInspector, useSource} from 'sanity'
+
+import {type PaneRouterContextValue, usePaneRouter} from '../../components'
+import {type PaneMenuItem} from '../../types'
+import {useStructureTool} from '../../useStructureTool'
+import {HISTORY_INSPECTOR_NAME, INSPECT_ACTION_PREFIX} from './constants'
+
+export function useDocumentPaneInspector({
+  documentId,
+  documentType,
+  params,
+}: {
+  params: NonNullable<PaneRouterContextValue['params']>
+  documentId: string
+  documentType: string
+}) {
+  const {setParams: setPaneParams} = usePaneRouter()
```

Comment:
Are the passed `params` also coming from `usePaneRouter()` ? Wondering if setParams could then be passed in as well? Or if we could get params from the hook